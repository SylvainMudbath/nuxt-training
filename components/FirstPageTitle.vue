<template>
  <h1 class="title">{{ titleTextHere }}</h1>
</template>

<script lang="ts">
import Vue, { PropOptions } from 'vue'

export default Vue.extend({
  name: 'FirstPageTitle',
  props: {
    titleTextHere: {
      type: String,
      required: true,
    } as PropOptions<string>,
  },
})
</script>

<style scoped lang="scss">
$bg-dark: #232323;

.title {
  color: $bg-dark;
}
</style>
