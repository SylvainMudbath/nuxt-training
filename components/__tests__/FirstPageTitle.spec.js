import { shallowMount, mount } from '@vue/test-utils'
import FirstPageTitle from '../FirstPageTitle.vue'

describe('FirstPageTitle', () => {
  test('renders correctly', () => {
    expect(
      mount(FirstPageTitle, {
        propsData: { titleTextHere: 'I am a title' },
      }).element
    ).toMatchSnapshot()
  })
})
