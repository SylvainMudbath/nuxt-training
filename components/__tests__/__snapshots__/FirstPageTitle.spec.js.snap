// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstPageTitle renders correctly 1`] = `
<h1
  class="title"
>
  I am a title
</h1>
`;
